org.ekstep.metadata-1.1/editor/directives/dialCode/dialCodeDirective.js

```diff
@@ -101,15 +101,18 @@ angular.module('editorApp', ['ngDialog', 'oc.lazyLoad', 'Scope.safeApply']).dire
         // clear dial code values
         $scope.clearDialCode = function () {
             $scope.dialcodes = "";
-            var nodeId = ecEditor.jQuery("#collection-tree").fancytree("getRootNode").getFirstChild().data.id;
-            if (org.ekstep.collectioneditor.cache.nodesModified && org.ekstep.collectioneditor.cache.nodesModified[nodeId]) {
-                org.ekstep.collectioneditor.cache.nodesModified[nodeId].metadata["dialcodes"] = [];
-            }
-            if (!stateService.state.dialCodeMap) {
-                stateService.create('dialCodeMap');
+            var currentNode = ecEditor.jQuery("#collection-tree").fancytree("getRootNode").getFirstChild().data;
+            _.has(stateService.state.invaliddialCodeMap, currentNode.id) ? _.unset(stateService.state.invaliddialCodeMap, currentNode.id) : "";
+            _.has(stateService.state.dialCodeMap, currentNode.id) ? _.unset(stateService.state.dialCodeMap, currentNode.id) : "";
+            if(currentNode.metadata && currentNode.metadata.dialcodes && currentNode.metadata.dialcodes.length){
+                if (org.ekstep.collectioneditor.cache.nodesModified && org.ekstep.collectioneditor.cache.nodesModified[currentNode.id]) {
+                    org.ekstep.collectioneditor.cache.nodesModified[currentNode.id].metadata["dialcodes"] = [];
+                }
+                if (!stateService.state.dialCodeMap) {
```

Comment:
This is not required. Verify & remove this code.

Reply:
After testing, we got to know this is required to save the un-linked(cleared) contents.